packages/cli-hydrogen/src/cli/commands/hydrogen/deploy.ts

```diff
@@ -0,0 +1,51 @@
+import {hydrogenFlags} from '../../flags.js'
+import {deployToOxygen} from '../../services/deploy.js'
+import {output, cli, environment, path} from '@shopify/cli-kit'
+import Command from '@shopify/cli-kit/node/base-command'
+import {Flags} from '@oclif/core'
+
+export default class Deploy extends Command {
+  static description = 'Deploy your Hydrogen app to Oxygen hosting'
+  static hidden = true
+
+  static flags = {
+    ...cli.globalFlags,
+    ...hydrogenFlags,
+    deploymentToken: Flags.string({
+      required: true,
+      env: 'SHOPIFY_HYDROGEN_FLAG_DEPLOYMENT_TOKEN',
+      description: 'Specify your Oxygen deployment token.',
+    }),
+    commitMessage: Flags.string({
+      env: 'SHOPIFY_HYDROGEN_FLAG_COMMIT_MESSAGE',
+      description: 'Override the default Git commit message.',
+    }),
+    commitAuthor: Flags.string({
+      env: 'SHOPIFY_HYDROGEN_FLAG_COMMIT_AUTHOR',
+      description: 'Override the default Git commit author.',
+    }),
+    healthCheck: Flags.boolean({
+      env: 'SHOPIFY_HYDROGEN_FLAG_HEALTH_CHECK',
+      default: true,
+      description: 'Require a health check before the deployment succeeds.',
+    }),
+    dmsAddress: Flags.string({
```

Comment:
In #443 where we create a separate "api service" to call DMS I changed it to be "Oxygen" as well as I think it made more sense (and we also don't need to change the name when we switch over to targeting `oxygen-hub`). I think keeping this flag for now is fine since it's hidden and descriptive of what it is actually doing.

Reply:
After some discussion over Slack I decided that it makes sense to use "Oxygen" language here as it will make our transition over to using `oxygen-hub` in the future easier and it also matches the new service name in #443